test/parallel/test-http-outgoing-proto.js

```diff
@@ -71,7 +71,7 @@ common.expectsError(() => {
   outgoingMessage.write('');
 }
 
-assert(OutgoingMessage.prototype.write.call({ _header: 'test' }));
+// assert(OutgoingMessage.prototype.write.call({ _header: 'test' }));
```

Comment:
What does this test? Doesn't seem like a valid case? @benjamingr

Reply:
This test was added in https://github.com/nodejs/node/commit/02ccffb081204c8f9b8b5a2a57354276f5676891 by @hiroppy. Looks like the goal was to provide test coverage for previously-uncovered code.

I agree that it doesn't seem like a valid test case. Instead of getting rid of it entirely, maybe change it to an `assert.throws()` so your type-checking on `chunk` is tested.

This PR may be semver-major because it throws in cases that did not throw previously.

Reply:
Oh, wait, I'm mistaken about the throw. It's already there. And the test case may in fact be valid at least in terms of having code go down a route that it did not previously. Will require closer investigation....

Reply:
Seems like there's a bug? This should throw but currently does not? Is that your impression, @ronag?

Reply:
> Seems like there's a bug? This should throw but currently does not? Is that your impression, @ronag?

Or is this wrong because `chunk` should not be required here?

Reply:
@Trottt: I think it should throw... which it does now... but didn't before.